### PR TITLE
Pr13 tests

### PR DESCRIPTION
**Pull request by IvanFadieiev**
Test coverage

### CODE REVIEW

**Review by OlegTuryk — APPROVED** [review]

File: src/test/java/project/bookstore/repository/CartItemRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            .setDescription("Description")
+            .setCoverImage("Cover image");
+
+    @Autowired
+    private CartItemsRepository cartItemsRepository;
+
+    @Test
+    @DisplayName("Verify getCartItemsByShoppingCartId() method works")
+    @Sql(scripts = {
+            "classpath:database/scripts/books/clear-books-table.sql",
```
Comment (vovasalyha) [inline src/test/java/project/bookstore/repository/CartItemRepositoryTest.java:38]: You have created constants for scripts in previous class, but here it is hardcoded. Try to be consistent.

File: src/test/java/project/bookstore/service/ShoppingCartServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+        when(shoppingCartRepository.getShoppingCartByUserId(VALID_ID))
+                .thenReturn(Optional.ofNullable(SHOPPING_CART));
+        when(cartItemsRepository.getCartItemsByShoppingCartId(VALID_ID))
+                .thenReturn(CART_ITEMS);
+        when(cartItemMapper.toDto(CART_ITEM)).thenReturn(CART_ITEM_RESPONSE);
+
+        ShoppingCartResponseDto shoppingCart = shoppingCartService.getShoppingCart();
+
+        assertEquals(shoppingCart, SHOPPING_CART_RESPONSE_DTO);
+        verify(shoppingCartMapper, Mockito.times(1))
```
Comment (vovasalyha) [inline src/test/java/project/bookstore/service/ShoppingCartServiceTest.java:132]: times(1) is default, so you don't need to specify this parameter. Additionally, using verify on stabbed method calls is redundant, you can check `verify` method documentation.

File: src/test/java/project/bookstore/service/ShoppingCartServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+                .getShoppingCartByUserId(VALID_ID);
+        verify(cartItemsRepository, Mockito.times(1))
+                .getCartItemsByShoppingCartId(VALID_ID);
+        verify(cartItemMapper, Mockito.times(1))
+                .toDto(CART_ITEM);
+    }
+
+    @Test
+    @DisplayName("Verify getShoppingCart() throws exception")
+    public void getShoppingCart_ThrowsException() {
```
Comment (vovasalyha) [inline src/test/java/project/bookstore/service/ShoppingCartServiceTest.java:146]: Ambiguous name. When should it throw an exception?

File: src/test/java/project/bookstore/service/ShoppingCartServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+                .getPrincipal();
+        verify(shoppingCartRepository, Mockito.times(1))
+                .getShoppingCartByUserId(VALID_ID);
+        verify(cartItemsRepository, Mockito.times(1))
+                .save(CART_ITEM_WITHOUT_ID);
+    }
+
+    @Test
+    @DisplayName("Verify addCartItemToShoppingCart() throws exception with invalid id")
+    public void addCartItemToShoppingCart_InValidBookId_ThrowsException() {
```
Comment (vovasalyha) [inline src/test/java/project/bookstore/service/ShoppingCartServiceTest.java:191]:
```suggestion
    public void addCartItemToShoppingCart_InvalidBookId_ThrowsException() {
```

File: src/test/java/project/bookstore/service/ShoppingCartServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+        assertEquals(actual, expected);
+        verify(AUTHENTICATION, Mockito.times(4))
+                .getPrincipal();
+        verify(shoppingCartRepository, Mockito.times(1))
+                .getShoppingCartByUserId(INVALID_ID);
+    }
+
+    @Test
+    @DisplayName("Verify addCartItemToShoppingCart() method works")
+    public void addCartItemToShoppingCart_ReturnsResponseStatusAccepted() {
```
Comment (vovasalyha) [inline src/test/java/project/bookstore/service/ShoppingCartServiceTest.java:169]: is it correct name for this test case `ReturnsResponseStatusAccepted`?

File: src/test/java/project/bookstore/service/ShoppingCartServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,308 @@
...
+                .getPrincipal();
+        verify(cartItemsRepository, Mockito.times(1))
+                .getCartItemByIdAndUserId(VALID_ID, VALID_ID);
+        verify(cartItemsRepository, Mockito.times(1))
+                .save(CART_ITEM);
+    }
+
+    @Test
+    @DisplayName("Verify updateCartItemQuantityById() throws exception with invalid id")
+    public void updateCartItemQuantityById_InValidId_ThrowsException() {
```
Comment (vovasalyha) [inline src/test/java/project/bookstore/service/ShoppingCartServiceTest.java:229]:
```suggestion
    public void updateCartItemQuantityById_InvalidId_ThrowsException() {
```

**Review by vovasalyha — CHANGES_REQUESTED** [review]
GJ, but see a couple of comments

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**